File: app/src/main/java/comx/y/z/kotlinbase/TestHolder.java
package comx.y.z.kotlinbase;

import android.view.ViewGroup;

import androidx.recyclerview.widget.RecyclerView;

import com.jibase.extensions.BindingExtensions;
import com.jibase.iflexible.adapter.FlexibleAdapter;
import com.jibase.iflexible.items.abstractItems.AbstractFlexibleItem;
import com.jibase.iflexible.items.binding.BindFlexibleItem;
import com.jibase.iflexible.viewholder.binding.BindFlexibleViewHolder;

import org.jetbrains.annotations.NotNull;

import java.util.List;

import comx.y.z.kotlinbase.databinding.ItemBinding;

public class TestHolder extends BindFlexibleItem {
    private String data;

    public TestHolder(String data) {
        this.data = data;
    }

    @Override
    public int getLayoutRes() {
        return R.layout.item;
    }


    @Override
    public void bindViewHolder(@NotNull FlexibleAdapter<?> adapter, @NotNull RecyclerView.ViewHolder holder, int position, @NotNull List<?> payloads) {
        ItemBinding binding = (ItemBinding) ((BindFlexibleViewHolder) holder).getBinding();
        binding.text.setText(data);
    }
}

File: app/src/main/java/comx/y/z/kotlinbase/TestHolder.kt
package comx.y.z.kotlinbase

import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.jibase.extensions.inflate
import com.jibase.iflexible.adapter.FlexibleAdapter
import com.jibase.iflexible.items.abstractItems.AbstractFlexibleItem
import com.jibase.iflexible.viewholder.FlexibleViewHolder
import kotlinx.android.synthetic.main.item_test.view.*

class TestHolder(val item: String) : AbstractFlexibleItem<TestHolder.ViewHolder>() {
    override fun bindViewHolder(adapter: FlexibleAdapter<*>, holder: RecyclerView.ViewHolder, position: Int, payloads: List<*>) {
        holder.itemView.text.text = item
    }

    override fun createViewHolder(parent: ViewGroup, adapter: FlexibleAdapter<*>): ViewHolder {
        return ViewHolder(parent.inflate(getLayoutRes()), adapter)
    }

    override fun getLayoutRes(): Int = R.layout.item_test

    class ViewHolder(preItemView: View, adapter: FlexibleAdapter<*>, isStickyHeader: Boolean = false) : FlexibleViewHolder(preItemView, adapter, isStickyHeader) {

    }
}